src/main/java/au/com/southsky/jfreesane/SaneDevice.java

```diff
@@ -214,4 +217,41 @@ SaneSession getSession() {
   void invalidateOptions() {
     optionTitleMap = null;
   }
+
+  @Override
+  public int hashCode() {
+    int hash = 3;
+    hash = 37 * hash + Objects.hashCode(this.name);
+    hash = 37 * hash + Objects.hashCode(this.vendor);
+    hash = 37 * hash + Objects.hashCode(this.model);
+    hash = 37 * hash + Objects.hashCode(this.type);
+    return hash;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
```

Comment:
> I think SaneDevice and SaneOption cannot readily be serialized, because they both have SaneSession fields. SaneSession in turn has a socket to the SANE server, so it can't be sensibly serialized.

How about seralizing everything of `SaneSession` except the socket and the socket address and port therefore?